From 4f9fca87ac436b8047b8871d2cc88778ffc7a3dc Mon Sep 17 00:00:00 2001
From: Jouni Malinen <j@w1.fi>
Date: Sun, 4 Mar 2012 16:13:00 +0200
Subject: [PATCH 54/58] Fix memory leak on parsing multiple network block eap
 entries

Signed-hostap: Jouni Malinen <j@w1.fi>
intended-for: hostap-1
(cherry picked from commit bb8b1bb0fcdf61511fb777cbe30a6b71d16ce4ce)
---
 wpa_supplicant/config.c |    1 +
 1 file changed, 1 insertion(+)

--- a/wpa_supplicant/config.c
+++ b/wpa_supplicant/config.c
@@ -1116,6 +1116,7 @@ static int wpa_config_parse_eap(const st
 
 	wpa_hexdump(MSG_MSGDUMP, "eap methods",
 		    (u8 *) methods, num_methods * sizeof(*methods));
+	os_free(ssid->eap.eap_methods);
 	ssid->eap.eap_methods = methods;
 	return errors ? -1 : 0;
 }
